real_estate/real_estate/doctype/property_unit/property_unit.py

```diff
@@ -20,3 +32,15 @@ def set_status(self, update=False, status=None, update_modified=False):
 
 		if update:
 			self.db_set('booking_status', self.booking_status, update_modified=update_modified)
+
+	def set_unit_template_booked_by(self, update=False, status=None, update_modified=False):
+		bookings = frappe.get_all('Property Booking Order', {'property_unit': self.name, 'docstatus': 1})
+
+		if bookings and self.booking_status == 'Booked':
```

Comment:
Why are you checking for booking_status?

Reply:
Just making sure its not already booked, but it work either ways so refactored